<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SudokuGame</class>
 <widget class="QMainWindow" name="SudokuGame">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>900</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>900</width>
    <height>900</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Sudoku</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QMainWindow {
    background-color: #1a1a2e;
}

QLabel#titleLabel {
    color: #eaf0f1;
    font-size: 32px;
    font-weight: bold;
    padding: 10px;
}

QLabel#statusLabel {
    color: #a4b0be;
    font-size: 14px;
    padding: 5px;
}

QPushButton {
    background-color: #0f3460;
    color: #eaf0f1;
    border: none;
    border-radius: 8px;
    padding: 12px 24px;
    font-size: 14px;
    font-weight: bold;
    min-width: 120px;
}

QPushButton:hover {
    background-color: #16537e;
}

QPushButton:pressed {
    background-color: #0a2647;
}

QPushButton#solveButton {
    background-color: #e94560;
}

QPushButton#solveButton:hover {
    background-color: #ff6b6b;
}

QPushButton#solveButton:pressed {
    background-color: #c73e54;
}

QPushButton#newGameButton {
    background-color: #00b894;
}

QPushButton#newGameButton:hover {
    background-color: #00d9a5;
}

QPushButton#newGameButton:pressed {
    background-color: #00a381;
}

QPushButton#clearButton {
    background-color: #fdcb6e;
    color: #2d3436;
}

QPushButton#clearButton:hover {
    background-color: #ffeaa7;
}

QPushButton#clearButton:pressed {
    background-color: #e5b85c;
}

QComboBox {
    background-color: #0f3460;
    color: #eaf0f1;
    border: none;
    border-radius: 8px;
    padding: 10px 15px;
    font-size: 14px;
    min-width: 150px;
}

QComboBox:hover {
    background-color: #16537e;
}

QComboBox::drop-down {
    border: none;
    width: 30px;
}

QComboBox::down-arrow {
    image: none;
    border-left: 5px solid transparent;
    border-right: 5px solid transparent;
    border-top: 8px solid #eaf0f1;
    margin-right: 10px;
}

QComboBox QAbstractItemView {
    background-color: #0f3460;
    color: #eaf0f1;
    selection-background-color: #16537e;
    border: none;
    border-radius: 8px;
}

QFrame#gridFrame {
    background-color: #16213e;
    border-radius: 15px;
    padding: 15px;
}

QWidget#numberPadWidget {
    background-color: transparent;
}</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="mainLayout">
    <property name="spacing">
     <number>15</number>
    </property>
    <property name="leftMargin">
     <number>30</number>
    </property>
    <property name="topMargin">
     <number>20</number>
    </property>
    <property name="rightMargin">
     <number>30</number>
    </property>
    <property name="bottomMargin">
     <number>20</number>
    </property>
    <item>
     <layout class="QHBoxLayout" name="headerLayout">
      <item>
       <widget class="QLabel" name="titleLabel">
        <property name="text">
         <string>Sudoku</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="controlsLayout">
      <property name="spacing">
       <number>15</number>
      </property>
      <item>
       <widget class="QPushButton" name="newGameButton">
        <property name="text">
         <string>New Game</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="difficultyCombo">
        <item>
         <property name="text">
          <string>Easy</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Medium</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Hard</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <spacer name="controlsSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="loadButton">
        <property name="text">
         <string>Load</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="saveButton">
        <property name="text">
         <string>Save</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QFrame" name="gridFrame">
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <layout class="QVBoxLayout" name="gridFrameLayout">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>15</number>
       </property>
       <property name="topMargin">
        <number>15</number>
       </property>
       <property name="rightMargin">
        <number>15</number>
       </property>
       <property name="bottomMargin">
        <number>15</number>
       </property>
       <item alignment="Qt::AlignHCenter">
        <widget class="QWidget" name="gridContainer" native="true">
         <property name="minimumSize">
          <size>
           <width>450</width>
           <height>450</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>450</width>
           <height>450</height>
          </size>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="numberPadWidget" native="true">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>60</height>
       </size>
      </property>
      <layout class="QHBoxLayout" name="numberPadLayout">
       <property name="spacing">
        <number>8</number>
       </property>
       <property name="leftMargin">
        <number>50</number>
       </property>
       <property name="topMargin">
        <number>5</number>
       </property>
       <property name="rightMargin">
        <number>50</number>
       </property>
       <property name="bottomMargin">
        <number>5</number>
       </property>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="actionLayout">
      <property name="spacing">
       <number>15</number>
      </property>
      <item>
       <spacer name="actionSpacerLeft">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="clearCellButton">
        <property name="text">
         <string>Clear Cell</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="clearBoardButton">
        <property name="text">
         <string>Clear Board</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="hintButton">
        <property name="text">
         <string>Hint</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="checkButton">
        <property name="text">
         <string>Check</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="solveButton">
        <property name="text">
         <string>Solve</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="actionSpacerRight">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QLabel" name="statusLabel">
      <property name="text">
       <string>Select a cell and enter a number</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>700</width>
     <height>24</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionLoad"/>
    <addaction name="actionSave"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuGame">
    <property name="title">
     <string>Game</string>
    </property>
    <addaction name="actionSolve"/>
    <addaction name="actionCheck"/>
    <addaction name="actionClear"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuGame"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionNew">
   <property name="text">
    <string>New Game</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionLoad">
   <property name="text">
    <string>Load Puzzle...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save Puzzle...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionSolve">
   <property name="text">
    <string>Solve</string>
   </property>
  </action>
  <action name="actionCheck">
   <property name="text">
    <string>Check Solution</string>
   </property>
  </action>
  <action name="actionClear">
   <property name="text">
    <string>Clear Board</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
